simd/src/Kokkos_SIMD_Common.hpp

```diff
@@ -354,6 +354,7 @@ template <class T, class BinaryOperation>
 struct Identity {
   KOKKOS_FORCEINLINE_FUNCTION
   operator T() {
+    // NOLINTNEXTLINE(bugprone-branch-clone)
```

Comment:
@ldh4 @tretre91 
This could also be rewritten as
```C++
   if constexpr (std::is_same_v<BinaryOperation, std::multiplies<>>) {
      return T(1);
    } else if constexpr (std::is_same_v<BinaryOperation, std::bit_and<>>) {
      return T(~T());
    } else {
      return T();
    }
```
Let me know if you prefer the suppression or if you want to refactor.

Reply:
For now, let's just place in a suppression.
I just recalled that there's a minor bug with simd reduction and identity element, and this struct may need a bit of adjustment anyway when fixing the bug.